version: '3.8'

services:
  backend:
    build: 
      context: ./backend
      dockerfile: Dockerfile
    command: gunicorn locateme.wsgi:application --bind 0.0.0.0:8000 --workers 4
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000/health/"]
      interval: 30s
      timeout: 10s
    volumes:
      - static_volume:/app/staticfiles
    env_file:
      - .env.docker
    deploy:
      resources:
        limits:
          cpus: '1.5'
          memory: 1G
    depends_on:
      db:
        condition: service_healthy
    networks:
      app-network:
        aliases:
          - backend.local

  db:
    image: postgres:15-alpine
    environment:
      POSTGRES_DB: ${DB_NAME}
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_HOST_AUTH_METHOD: scram-sha-256
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER}"]
      interval: 10s
    networks:
      - app-network

  frontend:
    build: 
      context: ./frontend
      dockerfile: Dockerfile
    environment:
      VITE_API_URL: http://backend.local:8000  # Используем алиас
    ports:
      - "5173:5173"
    networks:
      - app-network

volumes:
  postgres_data:
  static_volume:

networks:
  app-network:
    driver: bridge